### PR TITLE
Align iOS and Android Replace SDLSyncMsgVersion with SDLMsgVersion

### LINKED ISSUE #1352: Align SDLSyncMsgVersion with Android
### Bug Report
The iOS team uses `syncMsgVersion` in the RAI response, while the Android team uses `SdlMsgVersion`

### PR DESCRIPTION
**Pull request by justingluck93**
Fixes #1352 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Run app and check if responses and requests send proper data

### Summary
Deprecated SDLSyncMsgVersion and replaced it with SDLMsgVersion.  

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLRegisterAppInterfaceResponse.m [hunk 1]
```diff
@@ -31,11 +32,30 @@ - (instancetype)init {
 #pragma clang diagnostic pop
 
 - (void)setSyncMsgVersion:(nullable SDLSyncMsgVersion *)syncMsgVersion {
-    [self.parameters sdl_setObject:syncMsgVersion forName:SDLRPCParameterNameSyncMessageVersion];
+    if (syncMsgVersion == nil) {
+        [self.parameters sdl_setObject:nil forName:SDLRPCParameterNameSyncMessageVersion];
+        return;
+    }
+    SDLMsgVersion * sdlMsgVersion = [[SDLMsgVersion alloc] initWithMajorVersion:(uint8_t)[syncMsgVersion.majorVersion unsignedIntValue] minorVersion:(uint8_t)[syncMsgVersion.minorVersion unsignedIntValue] patchVersion:(uint8_t)[syncMsgVersion.patchVersion unsignedIntValue]];
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRegisterAppInterfaceResponse.m]:
```suggestion

    SDLMsgVersion *sdlMsgVersion = [[SDLMsgVersion alloc] initWithMajorVersion:(uint8_t)syncMsgVersion.majorVersion.unsignedIntValue minorVersion:(uint8_t)syncMsgVersion.minorVersion.unsignedIntValue patchVersion:(uint8_t)syncMsgVersion.patchVersion.unsignedIntValue];
```

File: SmartDeviceLink/SDLRegisterAppInterfaceResponse.m [hunk 2]
```diff
@@ -31,11 +32,30 @@ - (instancetype)init {
...
+    [self.parameters sdl_setObject:sdlMsgVersion forName:SDLRPCParameterNameSyncMessageVersion];
 }
 
 - (nullable SDLSyncMsgVersion *)syncMsgVersion {
-    return [self.parameters sdl_objectForName:SDLRPCParameterNameSyncMessageVersion ofClass:SDLSyncMsgVersion.class error:nil];
+    SDLMsgVersion * sdlMsgVersion = [self.parameters sdl_objectForName:SDLRPCParameterNameSyncMessageVersion ofClass:SDLMsgVersion.class error:nil];
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRegisterAppInterfaceResponse.m]:
```suggestion
    SDLMsgVersion *sdlMsgVersion = [self.parameters sdl_objectForName:SDLRPCParameterNameSyncMessageVersion ofClass:SDLMsgVersion.class error:nil];
```

File: SmartDeviceLink/SDLAppServiceManifest.m [hunk 3]
```diff
@@ -32,18 +33,28 @@ - (instancetype)initWithAppServiceType:(SDLAppServiceType)serviceType {
...
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 - (instancetype)initWithMediaServiceName:(nullable NSString *)serviceName serviceIcon:(nullable SDLImage *)serviceIcon allowAppConsumers:(BOOL)allowAppConsumers rpcSpecVersion:(nullable SDLSyncMsgVersion *)rpcSpecVersion handledRPCs:(nullable NSArray<NSNumber<SDLInt> *> *)handledRPCs mediaServiceManifest:(nullable SDLMediaServiceManifest *)mediaServiceManifest {
     return [self initWithServiceName:serviceName serviceType:SDLAppServiceTypeMedia serviceIcon:serviceIcon allowAppConsumers:allowAppConsumers rpcSpecVersion:rpcSpecVersion handledRPCs:handledRPCs mediaServiceManifest:mediaServiceManifest weatherServiceManifest:nil navigationServiceManifest:nil];
 }
+#pragma clang diagnostic pop
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLAppServiceManifest.m]: This is missing a `pop`

File: SmartDeviceLink/SDLAppServiceManifest.m [hunk 4]
```diff
@@ -95,12 +137,33 @@ - (void)setAllowAppConsumers:(nullable  NSNumber<SDLBool> *)allowAppConsumers {
...
+        [self.store sdl_setObject:nil forName:SDLRPCParameterNameRPCSpecVersion];
+        return;
+    }
+    SDLMsgVersion *maxRPCSpecVersion = [[SDLMsgVersion alloc] initWithMajorVersion:(uint8_t)[rpcSpecVersion.majorVersion unsignedIntValue] minorVersion:(uint8_t)[rpcSpecVersion.minorVersion unsignedIntValue] patchVersion:(uint8_t)[rpcSpecVersion.patchVersion unsignedIntValue]];
+    [self.store sdl_setObject:maxRPCSpecVersion forName:SDLRPCParameterNameRPCSpecVersion];
 }
 
 - (nullable SDLSyncMsgVersion *)rpcSpecVersion {
-    return [self.store sdl_objectForName:SDLRPCParameterNameRPCSpecVersion ofClass:SDLSyncMsgVersion.class error:nil];
+    SDLMsgVersion * maxRPCSpecVersion = [self.store sdl_objectForName:SDLRPCParameterNameRPCSpecVersion ofClass:SDLMsgVersion.class error:nil];
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLAppServiceManifest.m]:
```suggestion
    SDLMsgVersion *maxRPCSpecVersion = [self.store sdl_objectForName:SDLRPCParameterNameRPCSpecVersion ofClass:SDLMsgVersion.class error:nil];
```

File: SmartDeviceLink/SDLAppServiceManifest.m [hunk 5]
```diff
@@ -95,12 +137,33 @@ - (void)setAllowAppConsumers:(nullable  NSNumber<SDLBool> *)allowAppConsumers {
...
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 - (void)setRpcSpecVersion:(nullable SDLSyncMsgVersion *)rpcSpecVersion {
-    [self.store sdl_setObject:rpcSpecVersion forName:SDLRPCParameterNameRPCSpecVersion];
+    if (rpcSpecVersion == nil) {
+        [self.store sdl_setObject:nil forName:SDLRPCParameterNameRPCSpecVersion];
+        return;
+    }
+    SDLMsgVersion *maxRPCSpecVersion = [[SDLMsgVersion alloc] initWithMajorVersion:(uint8_t)[rpcSpecVersion.majorVersion unsignedIntValue] minorVersion:(uint8_t)[rpcSpecVersion.minorVersion unsignedIntValue] patchVersion:(uint8_t)[rpcSpecVersion.patchVersion unsignedIntValue]];
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLAppServiceManifest.m]:
```suggestion
    SDLMsgVersion *maxRPCSpecVersion = [[SDLMsgVersion alloc] initWithMajorVersion:(uint8_t)rpcSpecVersion.majorVersion.unsignedIntValue minorVersion:(uint8_t)rpcSpecVersion.minorVersion.unsignedIntValue patchVersion:(uint8_t)rpcSpecVersion.patchVersion.unsignedIntValue];
```

File: SmartDeviceLink/SDLAppServiceManifest.m [hunk 6]
```diff
@@ -95,12 +137,33 @@ - (void)setAllowAppConsumers:(nullable  NSNumber<SDLBool> *)allowAppConsumers {
...
 }
 
 - (nullable SDLSyncMsgVersion *)rpcSpecVersion {
-    return [self.store sdl_objectForName:SDLRPCParameterNameRPCSpecVersion ofClass:SDLSyncMsgVersion.class error:nil];
+    SDLMsgVersion * maxRPCSpecVersion = [self.store sdl_objectForName:SDLRPCParameterNameRPCSpecVersion ofClass:SDLMsgVersion.class error:nil];
+
+    if(maxRPCSpecVersion == nil) {
+        return [self.store sdl_objectForName:SDLRPCParameterNameRPCSpecVersion ofClass:SDLSyncMsgVersion.class error:nil];
+    }
+    return [[SDLSyncMsgVersion alloc] initWithMajorVersion:(uint8_t)[maxRPCSpecVersion.majorVersion unsignedIntValue] minorVersion:(uint8_t)[maxRPCSpecVersion.minorVersion unsignedIntValue] patchVersion:(uint8_t)[maxRPCSpecVersion.patchVersion unsignedIntValue]];
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLAppServiceManifest.m]:
```suggestion

    return [[SDLSyncMsgVersion alloc] initWithMajorVersion:(uint8_t)maxRPCSpecVersion.majorVersion.unsignedIntValue minorVersion:(uint8_t)maxRPCSpecVersion.minorVersion.unsignedIntValue patchVersion:(uint8_t)maxRPCSpecVersion.patchVersion.unsignedIntValue];
```

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLRegisterAppInterfaceResponseSpec.m [hunk 7]
```diff
@@ -68,61 +78,108 @@
...
+                                 SDLRPCParameterNameIconResumed: @YES,
+                                 },
+                           SDLRPCParameterNameOperationName:SDLRPCFunctionNameRegisterAppInterface}};
+        });
+
+        it(@"Should get correctly when initialized with a dictionary and get syncMsgVersion first", ^ {
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-        SDLRegisterAppInterfaceResponse* testResponse = [[SDLRegisterAppInterfaceResponse alloc] initWithDictionary:dict];
+            SDLRegisterAppInterfaceResponse* testResponse = [[SDLRegisterAppInterfaceResponse alloc] initWithDictionary:dict];
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLRegisterAppInterfaceResponseSpec.m]:
```suggestion
        SDLRegisterAppInterfaceResponse* testResponse = [[SDLRegisterAppInterfaceResponse alloc] initWithDictionary:dict];
```

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m [hunk 8]
```diff
@@ -40,20 +45,30 @@
...
         testWeatherServiceManifest = [[SDLWeatherServiceManifest alloc] initWithCurrentForecastSupported:true maxMultidayForecastAmount:3 maxHourlyForecastAmount:0 maxMinutelyForecastAmount:0 weatherForLocationSupported:false];
         testMediaServiceManifest = [[SDLMediaServiceManifest alloc] init];
         testNavigationServiceManifest = [[SDLNavigationServiceManifest alloc] init];
     });
 
     it(@"Should set and get correctly", ^{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
         SDLAppServiceManifest *testStruct = [[SDLAppServiceManifest alloc] init];
+#pragma clang diagnostic pop
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m]: This push pop is unnecessary

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m [hunk 9]
```diff
@@ -63,38 +78,77 @@
...
+        testStruct.maxRPCSpecVersion = testSDLMsgVersion;
+        expect(testStruct.maxRPCSpecVersion).to(equal(testSDLMsgVersion));
+        expect(testStruct.maxRPCSpecVersion).to(equal(testSDLMsgVersion));
         expect(testStruct.handledRPCs).to(equal(testHandledRPCs));
         expect(testStruct.weatherServiceManifest).to(equal(testWeatherServiceManifest));
         expect(testStruct.mediaServiceManifest).to(equal(testMediaServiceManifest));
         expect(testStruct.navigationServiceManifest).to(equal(testNavigationServiceManifest));
     });
+    describe(@"Test Dict", ^{
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m]:
```suggestion
    describe(@"test initializing with dictionary", ^{
```

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m [hunk 10]
```diff
@@ -63,38 +78,77 @@
...
-                               SDLRPCParameterNameHandledRPCs:testHandledRPCs,
-                               SDLRPCParameterNameWeatherServiceManifest:testWeatherServiceManifest,
-                               SDLRPCParameterNameMediaServiceManifest:testMediaServiceManifest,
-                               SDLRPCParameterNameNavigationServiceManifest:testNavigationServiceManifest
-                               };
+        it(@"Should get correctly when initialized with a dictionary and using SDLMsgVersion", ^{
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-        SDLAppServiceManifest *testStruct = [[SDLAppServiceManifest alloc] initWithDictionary:dict];
+            SDLAppServiceManifest *testStruct = [[SDLAppServiceManifest alloc] initWithDictionary:dict];
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m]:
```suggestion
        SDLAppServiceManifest *testStruct = [[SDLAppServiceManifest alloc] initWithDictionary:dict];
```

File: SmartDeviceLink/SDLMsgVersion.h [hunk 11]
```diff
@@ -0,0 +1,43 @@
+//
+//  SDLMsgVersion.h
+//  SmartDeviceLink
+//
+//  Created by Justin Gluck on 7/18/19.
+//  Copyright © 2019 smartdevicelink. All rights reserved.
+//
+
+#import "SDLRPCMessage.h"
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLMsgVersion.h]: fixed

File: SmartDeviceLink/SDLMsgVersion.h [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+/**
+ * Specifies the version number of the SDL V4 interface. This is used by both the application and SDL to declare what interface version each is using.
+ *
+ * @since SDL 1.0
+ */
+@interface SDLMsgVersion : SDLRPCStruct
+
+- (instancetype)initWithMajorVersion:(UInt8)majorVersion minorVersion:(UInt8)minorVersion patchVersion:(UInt8)patchVersion;
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLMsgVersion.h]: added

File: SmartDeviceLink/SDLMsgVersion.h [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * The major version indicates versions that is not-compatible to previous versions
+ *
+ * Required, Integer, 1 - 10
+ */
+@property (strong, nonatomic) NSNumber<SDLInt> *majorVersion;
+/**
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLMsgVersion.h]: done

File: SmartDeviceLink/SDLAppServiceManifest.m (hunk 3 above)
Comment (justingluck93) [reply SmartDeviceLink/SDLAppServiceManifest.m]: added

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m (hunk 8 above)
Comment (justingluck93) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m]: removed

File: SmartDeviceLink/SDLSyncMsgVersion.h [hunk 14]
```diff
@@ -11,6 +11,7 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * @since SDL 1.0
  */
+DEPRECATED_MSG_ATTRIBUTE("Use SDLMsgVersion instead")
```
[the remark this answers is not in the document]
Comment (justingluck93) [reply SmartDeviceLink/SDLSyncMsgVersion.h]: changed to this

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m (hunk 8 above)
Comment (justingluck93) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLAppServiceManifestSpec.m]: removed

**Comment by joeljfischer** [comment]
@justingluck93 Conflicts have appeared

**Pull request merged**